.github/workflows/check-whitespaces.yaml

```diff
@@ -0,0 +1,51 @@
+name: Check trailing white spaces before merge
+
+on:
+  pull_request:
+  push:
+    branches: [ "main" ]
+    paths:
```

Comment:
I merged too fast, this needs to be
```
pull_request:
    branches:
      - main
```
otherwise the CI will only be executed when pushing on main, so after the PR is merged

Reply:
You sure? the workflow ran for me just now on this PR
https://github.com/parodos-dev/serverless-workflows/pull/422/checks

Reply:
ah, indeed an empty `pull_request` means that every PR will execute the CI (https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request)